infra/src/main/kotlin/fi/fta/geoviite/infra/util/List.kt

```diff
@@ -23,6 +19,21 @@ fun rangesOfConsecutiveIndicesOf(
         .map { c -> c[0] until c[1] + offsetRangeEndsBy }
         .toList()
 
+/**
+ * "Compact list" = lists containing all values between their endpoints, in order. Generally this is meant for the
+ * address points of different alignments in the same geocoding context.
+ */
+fun <T : Comparable<T>> findCommonSubsequenceInCompactLists(a: List<T>, b: List<T>): Pair<IntRange, IntRange>? {
+    if (a.isEmpty() || b.isEmpty()) return null
+
+    val aInB = getIndexRangeForRangeInOrderedList(b, a.first(), a.last(), Comparator.naturalOrder<T>()::compare)
+    val bInA = getIndexRangeForRangeInOrderedList(a, b.first(), b.last(), Comparator.naturalOrder<T>()::compare)
+
+    return if (aInB != null && bInA != null) {
+        aInB to bInA
+    } else null
+}
```

Comment:
Tässä kanssa vähän samaa kuin mitä point-outtasin aiemmilla revieweillä, eli tämä on aika spesifi funktio yleiseksi listan helpperifunktioksi. Mietin tässäkin olisiko vaan järkevämpää sijoittaa tämä esim. geokoodauskontekstiin tai jonnekin muualle sinnepäin.